sdk/cosmos/azure_data_cosmos/src/retry_policies/resource_throttle_retry_policy.rs

```diff
@@ -0,0 +1,440 @@
+    ///   - `Err(azure_core::Error)` - A network or client-side error
+    ///
+    /// # Returns
+    ///
+    /// A `RetryResult` indicating the retry decision.
+    async fn should_retry(&self, response: &azure_core::Result<RawResponse>) -> RetryResult {
+        match response {
+            Ok(resp) => {
+                if resp.status().is_server_error() || resp.status().is_client_error() {
+                    return self.should_retry_response(resp);
+                }
+
+                RetryResult::DoNotRetry
+            }
+            Err(err) => self.should_retry_error(err),
+        }
+    }
+}
+
+// Tests for resource throttle retry policy
+#[cfg(test)]
+mod tests {
+    use crate::retry_policies::resource_throttle_retry_policy::ResourceThrottleRetryPolicy;
+    use crate::retry_policies::RetryPolicy;
+    use crate::retry_policies::RetryResult;
+    use azure_core::http::{RawResponse, StatusCode};
+    use azure_core::time::Duration;
+
+    /// Helper function to create a mock RawResponse with a given status code
+    fn create_mock_response(status: StatusCode) -> azure_core::Result<RawResponse> {
+        use typespec_client_core::http::headers::Headers;
+
+        // Create headers
+        let mut headers = Headers::new();
+        headers.insert("content-type", "application/json");
+
+        azure_core::Result::from(Ok(RawResponse::from_bytes(
+            status,
+            headers,
+            r#"{"id":12,"name":"Too Many Requests"}"#,
+        )))
+    }
+    #[tokio::test]
+    async fn test_retry_policy_handles_429_status() {
+        // Create a retry policy with 3 max retries, 100 second max wait time, and backoff factor of 2
+        let policy = ResourceThrottleRetryPolicy::new(3, 100, 2);
+
+        // Simulate a 429 TooManyRequests response
+        let response_429 = create_mock_response(StatusCode::TooManyRequests);
+
+        // First retry attempt
+        let result1 = policy.should_retry(&response_429).await;
+        match result1 {
+            RetryResult::Retry { after } => {
+                assert!(after > Duration::ZERO, "Should have backoff time");
+                println!("First retry - backoff time: {:?}", after);
+            }
+            RetryResult::DoNotRetry => panic!("Should retry on first 429 response"),
+        }
+
+        // Second retry attempt
+        let result2 = policy.should_retry(&response_429).await;
+        let backoff2 = match result2 {
+            RetryResult::Retry { after } => {
+                assert!(after > Duration::ZERO, "Should have backoff time");
+                after
+            }
+            RetryResult::DoNotRetry => panic!("Should retry on second 429 response"),
+        };
+
+        // Extract backoff1 for comparison
+        let backoff1 = match result1 {
+            RetryResult::Retry { after } => after,
+            _ => panic!("Expected retry result"),
+        };
```

Comment:
There are a few more compact ways to express "two-arm" matches (where one arm is just `_`). I'd suggest this:

```rust
let RetryResult::Retry { after: backoff1 } = result1 else {
    panic!("Expected retry result")
}
```

That will match on `RetryResult::Retry` and then bind the `after` field to the local variable `backoff1`. If the result doesn't match that pattern, the `else` block will be taken. It's the same code as you have above, just a slightly more compact and idiomatic way to write it.

There's also a similar inverted syntax:

```rust
let backoff1 = if let RetryResult::Retry { after } = value {
    after
} else {
    panic!("Expected retry result");
}
```

That one is just FYI, I think the first syntax is preferable in this case.